pkg/infrastructure/aws/aws.go

```diff
@@ -60,7 +60,7 @@ type output struct {
 }
 
 // Provision creates cluster infrastructure using AWS SDK calls.
-func (a InfraProvider) Provision(dir string, parents asset.Parents) ([]*asset.File, error) {
+func (a InfraProvider) Provision(_ context.Context, dir string, parents asset.Parents) ([]*asset.File, error) {
```

Comment:
What's the planned used for the context here?

Reply:
It can be used in this function to replace the `context.Background()` used but not worth doing it in this PR since the AWS SDK will become obsolete once aws capi is in place.

Reply:
Thanks, I understood it was just a temp change to cope with the patch. I was wondering if, in addition to support to clean shutdown, there was any other use for that within the capi provisioners (ie inject a piece of data)

Reply:
:+1: @andfasano I didn't have much else in mind besides the clean shutdown of the capi controller processes. You've already pointed out some good uses. It seems like there may be some opportunities to reduce some global scope.